runtime/jmxcontrol/src/main/java/quarks/runtime/jmxcontrol/JMXControlService.java

```diff
@@ -127,29 +126,40 @@ public void unregister(String controlId) {
 	}
 
     @Override
-    public <T> Set<T> getControls(Class<T> controlInterface) {
+    public <T> T  getControl(String type, String alias, Class<T> controlInterface) {
         try {
             MBeanServer mBeanServer = getMbs();
-            Set<ObjectName> names = getObjectNamesForInterface(controlInterface.getName());
+            Set<ObjectName> names = getObjectNamesForInterface(type, alias, controlInterface.getName());
+            
+            if (names.isEmpty())
+                return null;
+            if (names.size() != 1)
+                throw new RuntimeException("Alias " + alias + " not unique for type " + type);
```

Comment:
I'm just changing getControls() in this pull request. It only expects a single MBean registered, so if it finds more it complains.